### ISSUE TITLE
after I update vscode-insiders error ocurried 

### ISSUE THREAD

**Issue by GeorgiNiagulov** [robb0wen/synthwave-vscode#317]
![synthwaveError](https://github.com/user-attachments/assets/8fb16d40-b589-47bc-8208-15b55aefea60)

**Comment by VanderSP** [comment]
YESSSSSSSSSSSSSSSS also just happened now... was 1.94 update.... how i will code now! i always needed to reenable synthwave, in admin mode... then reopen... etc... but was worth! :D

PLEASE WE WANT FIX!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by vinayakkulkarni** [comment]
```sh
024-09-16 02:28:08.674 [error] TypeError: Cannot read properties of undefined (reading 'filename')
    at /Users/vinayak/.vscode-insiders/extensions/robbowen.synthwave-vscode-0.1.15/src/extension.js:27:44
```

Seems the VSCode insiders update introduced a bug on this line –> https://github.com/robb0wen/synthwave-vscode/blob/master/src/extension.js#L27

**Comment by vinayakkulkarni** [comment]
Found a possible solution / hack:

update your local `extension.js` with this:

```js
      const isWin = /^win/.test(process.platform);
      const appDir = `${path.dirname(vscode.env.appRoot)}/app/out`;
      const base = appDir + (isWin ? "\\vs\\code" : "/vs/code");
      const electronBase = isVSCodeBelowVersion("1.70.0")
        ? "electron-browser"
        : "electron-sandbox";
      const htmlFile =
        base +
        (isWin
          ? "\\" + electronBase + "\\workbench\\workbench.esm.html"
          : "/" + electronBase + "/workbench/workbench.esm.html");
      const templateFile =
        base +
        (isWin
          ? "\\" + electronBase + "\\workbench\\neondreams.js"
          : "/" + electronBase + "/workbench/neondreams.js");
```

Basically you have to patch these lines – https://github.com/robb0wen/synthwave-vscode/blob/master/src/extension.js#L26-L41

Might send a PR for the fix.

**Comment by VanderSP** [comment]
Hey THANKS! Back working in NEON!!!!!!!!!

**Comment by GeorgiNiagulov** [comment]
Yes this fix it's working thanks :) I believe https://github.com/robb0wen will see and accept pull request!

**Comment by jd78** [comment]
I'm experiencing the same problem on MacOS, and the suggested solution isn't effective for this operating system. After making some adjustments, I got that VS Code needs to run with administrative privileges. And even doing so nothing happens.

**Comment by ikariiwarrior** [comment]
This is now an issue on public release of VSCode. Nice work, @vinayakkulkarni for patching in a fix. Hopefully it will be rolled into master and deployed soon. :)

**Comment by badrat8** [comment]
@vinayakkulkarni thanks a lot, couldnt work without this theme :)

**Comment by AgGro00** [comment]
Same problem

**Comment by NathyVZM** [comment]
@vinayakkulkarni thank you so much for the patch 😭❤️ i can be happy again with the neon dreams. Hopefully they'll push this to `master` 🥹

**Comment by AgGro00** [comment]
> Found a possible solution / hack:
> 
> update your local `extension.js` with this:
> 
> ```js
>       const isWin = /^win/.test(process.platform);
>       const appDir = `${path.dirname(vscode.env.appRoot)}/app/out`;
>       const base = appDir + (isWin ? "\\vs\\code" : "/vs/code");
>       const electronBase = isVSCodeBelowVersion("1.70.0")
>         ? "electron-browser"
>         : "electron-sandbox";
>       const htmlFile =
>         base +
>         (isWin
>           ? "\\" + electronBase + "\\workbench\\workbench.esm.html"
>           : "/" + electronBase + "/workbench/workbench.esm.html");
>       const templateFile =
>         base +
>         (isWin
>           ? "\\" + electronBase + "\\workbench\\neondreams.js"
>           : "/" + electronBase + "/workbench/neondreams.js");
> ```
> 
> Basically you have to patch these lines – https://github.com/robb0wen/synthwave-vscode/blob/master/src/extension.js#L26-L41
> 
> Might send a PR for the fix.

It worked for me to replace the current configuration with this

**Comment by vinayakkulkarni** [comment]
Lol, my VSCode updated to v1.95 & it stopped working again, seems VSCode reverted `workbench.esm.html` to `workbench.html`. Updated the PR [`4a2935d` (#318)](https://github.com/robb0wen/synthwave-vscode/pull/318/commits/4a2935d49cb31770febf62f5e973b6996d14bdd8)

**Comment by ArcticOc** [comment]
It seems that `Fix Checksums: apply` is not working properly either.

**Comment by barguti** [comment]
> Found a possible solution / hack:
> 
> update your local `extension.js` with this:
> 
> ```js
>       const isWin = /^win/.test(process.platform);
>       const appDir = `${path.dirname(vscode.env.appRoot)}/app/out`;
>       const base = appDir + (isWin ? "\\vs\\code" : "/vs/code");
>       const electronBase = isVSCodeBelowVersion("1.70.0")
>         ? "electron-browser"
>         : "electron-sandbox";
>       const htmlFile =
>         base +
>         (isWin
>           ? "\\" + electronBase + "\\workbench\\workbench.esm.html"
>           : "/" + electronBase + "/workbench/workbench.esm.html");
>       const templateFile =
>         base +
>         (isWin
>           ? "\\" + electronBase + "\\workbench\\neondreams.js"
>           : "/" + electronBase + "/workbench/neondreams.js");
> ```
> 
> Basically you have to patch these lines – https://github.com/robb0wen/synthwave-vscode/blob/master/src/extension.js#L26-L41
> 
> Might send a PR for the fix.

It works, thanks @AgGro00

**Comment by johnelliottwizio** [comment]
it worked thank you @AgGro00 , my brain can't work wit out GLOW

**Comment by harishhbaskar** [comment]
I am on fedora linux will this fix work on my system?

**Comment by puentebravo** [comment]
> Found a possible solution / hack:
> 
> update your local `extension.js` with this:
> 
> ```js
>       const isWin = /^win/.test(process.platform);
>       const appDir = `${path.dirname(vscode.env.appRoot)}/app/out`;
>       const base = appDir + (isWin ? "\\vs\\code" : "/vs/code");
>       const electronBase = isVSCodeBelowVersion("1.70.0")
>         ? "electron-browser"
>         : "electron-sandbox";
>       const htmlFile =
>         base +
>         (isWin
>           ? "\\" + electronBase + "\\workbench\\workbench.esm.html"
>           : "/" + electronBase + "/workbench/workbench.esm.html");
>       const templateFile =
>         base +
>         (isWin
>           ? "\\" + electronBase + "\\workbench\\neondreams.js"
>           : "/" + electronBase + "/workbench/neondreams.js");
> ```
> 
> Basically you have to patch these lines – https://github.com/robb0wen/synthwave-vscode/blob/master/src/extension.js#L26-L41
> 
> Might send a PR for the fix.

You're a Neon saint - thank you! I live for that glow, man.

**Comment by UR-Hiro** [comment]
Can you teach me how to update it. I don't know how to do it. And I cannot live without the glow. 
Thanks, in advance.

**Comment by koudy2** [comment]
> I am on fedora linux will this fix work on my system?

Yes it works. Running fedora as well



> Can you teach me how to update it. I don't know how to do it. And I cannot live without the glow. Thanks, in advance.

What you see here (the piece of code pasted above in this issue) is the fix.
1) Navigate to `$USER/.vscode/extensions/robbowen.synthwave-.../src` (note that ... before /src is placeholder for string with version) and there will be extension.js
2) Make backup copy of this file so you have something to fall back to in case you need to start over
3) Locate lines 26-41
4) Paste the code from here to replace them and save the file

**Comment by larrydevofficial** [comment]
Working on Mac with latest VSCode 1.94.2. After enabling neon dreams, you might have to go to

> Preferences: Color Theme

scroll down and hover over SynthWave '84 then you should be able to see the lights when you select your actual preferred Color Theme

**Comment by alex1727tonato** [comment]
> > Estoy en Fedora Linux. ¿Esta solución funcionará en mi sistema?
> 
> Sí, funciona. También se ejecuta Fedora.
> 
> > ¿Puedes enseñarme cómo actualizarlo? No sé cómo hacerlo y no puedo vivir sin el brillo. Gracias de antemano.
> 
> Lo que ves aquí (el fragmento de código pegado arriba en este problema) es la solución.
> 
> 1. Vaya a `$USER/.vscode/extensions/robbowen.synthwave-.../src`(tenga en cuenta que ... antes de /src hay un marcador de posición para la cadena con la versión) y habrá extension.js
> 2. Haz una copia de seguridad de este archivo para tener algo a lo que recurrir en caso de que necesites empezar de nuevo.
> 3. Localiza las líneas 26-41
> 4. Pegue el código desde aquí para reemplazarlos y guardar el archivo.

For Ubuntu, where is?

**Comment by robb0wen** [comment]
This issue should now be fixed in v0.1.16

**Comment by AgGro00** [comment]
> it worked thank you @AgGro00 , my brain can't work wit out GLOW

haha sameee
